fe/fe-core/src/main/java/com/starrocks/analysis/BinaryPredicate.java

```diff
@@ -192,16 +193,8 @@ public static Type getCmpType(Type type1, Type type2) {
         if (type1.isJsonType() || type2.isJsonType()) {
             return Type.JSON;
         }
-        if (type1.isArrayType() || type2.isArrayType()) {
-            // Must both be array
-            if (!type1.matchesType(type2)) {
-                return Type.INVALID;
-            }
-            return type1;
-        }
         if (type1.isComplexType() || type2.isComplexType()) {
-            // We don't support complex type for binary predicate.
-            return Type.INVALID;
+            return TypeManager.getCommonSuperType(type1, type2);
```

Comment:
can't modify here, we only support equals, the method will check `><` too

Reply:
disable the function

Reply:
Is it expected? maybe we shouldn't change this function

Reply:
just allow  array type to do any binary predicts, map/struct only do equal binary predicts.